### ISSUE TITLE
Support setting a Host return type in Component contract

### ISSUE THREAD

**Issue by efoken** [atomicojs/atomico#82]
IMO the `Component` type should also support defining a ReturnType, like we can do when using `function componentName(props: Props<typeof props>): Host<{}>`

Should be something like this:

```ts
export type Component<props = null, meta = any> = props extends null
  ? {
      (props: FillObject): Host<meta>;
      props?: SchemaProps;
      styles?: Sheets;
    }
  : {
      (props: DOMProps<props>): Host<meta>;
      props: SchemaInfer<props> & MetaProps<Partial<props>>;
      styles?: Sheets;
    };
```

**Comment by UpperCod** [comment]
Thanks @efoken , Good idea, now in atomico@1.61.0 setting meta props is supported in the way you comment https://github.com/atomicojs/atomico/releases/tag/1.61.0 

I am attentive if the shared version solves your issue.

---

By the way, on the [Atomico site](https://atomicojs.dev/) I plan to add the following block:

![Web 1280 – 14](https://user-images.githubusercontent.com/9159849/179339182-cc15b871-18cb-44e8-b6f9-67f8485ffb41.png)

I would be very happy if you have any project, ideal or Company in which you have applied Atomico (Only if it can be public) with this I hope to further encourage the use of Atomico 😊

**Comment by efoken** [comment]
Thanks for changing this, but now I have another problem: For all components that use the `Component` contract, all other DOM properties (like `aria-*`, `onclick`, `onkeydown`, etc.) are missing in type definitions.

<img width="723" alt="Screenshot 2022-07-18 at 18 06 55" src="https://user-images.githubusercontent.com/522446/179554476-a2544a62-5218-45f1-b3bd-bb54ca88dc3d.png">

**Comment by efoken** [comment]
> By the way, on the [Atomico site](https://atomicojs.dev/) I plan to add the following block:
> 
> ![Web 1280 – 14](https://user-images.githubusercontent.com/9159849/179339182-cc15b871-18cb-44e8-b6f9-67f8485ffb41.png)
> 
> I would be very happy if you have any project, ideal or Company in which you have applied Atomico (Only if it can be public) with this I hope to further encourage the use of Atomico 😊

I'm working at IBM iX and we are using Atomico for the client [Barmer](http://barmer.de/) for building everything new from the ground up. We have React Native (Web) components and also WebComponents using Atomico and share a lot of code like styles, helpers and hooks, and we already launched the new app, based on React Native – new website is following @ some day, which will be based on WebComponents 🙂

- public Storybook URL for the WebComponents: https://barmer-puls-html.entw.bconnect.barmer.de/
- Website based on Next.js and React Native Web is located here: https://barmer-puls.entw.bconnect.barmer.de/

We are also currently working on a public website for the Design System

**Comment by UpperCod** [comment]
**regarding the bug**

Ready fixed in version 1.61.1.  

**regarding the use case**

I am really happy with the news😊, I hope to be able to share all this on the Atomico site,  can I reference your use case on the Atomico site?
